packages/synapse-react-client/src/utils/AppUtils/AppUtils.ts

```diff
@@ -1,11 +1,39 @@
 import { useHistory } from 'react-router-dom'
 import { LAST_PLACE_LOCALSTORAGE_KEY } from '../SynapseConstants'
 import { useEffect, useState } from 'react'
+import UniversalCookies from 'universal-cookie'
 
 export function storeLastPlace() {
   // save current route (so that we can go back here after SSO)
   localStorage.setItem(LAST_PLACE_LOCALSTORAGE_KEY, window.location.href)
 }
+const cookies = new UniversalCookies()
+export const ONE_SAGE_REDIRECT_COOKIE_KEY =
+  'org.sagebionetworks.cookies.redirect-after-login'
+
+export function storeRedirectURLForOneSageLogin() {
+  // save current URL in a cookie that One Sage will use to send you back to the correct page
+  const domainValue = window.location.hostname
+    .toLowerCase()
+    .endsWith('.synapse.org')
+    ? '.synapse.org'
+    : undefined
+  cookies.set(ONE_SAGE_REDIRECT_COOKIE_KEY, window.location.href, {
+    path: '/',
+    domain: domainValue,
+  })
```

Comment:
I was thinking we should expire the cookie, but the only case I think that would matter is this sequence:

1. I am redirected to log in from App A, which sets the cookie
2. I do not complete authentication for whatever reason
3. After a while, I use App B, which does NOT set the cookie and it redirects me to log in
4. I log in to Synapse and it sends me to App A

Kind of an edge case, WDYT? IMO it doesn't hurt to have it expire after a few hours.

Reply:
No harm in setting the expiration.  The "session" cookie could be deleted by that time, and App B (all of our apps) _should_ set the cookie, but it's a good idea to guard against this case.  Thanks!